lib/read_parsers.cc

```diff
@@ -63,10 +63,29 @@ void SeqAnParser::imprint_next_read(Read &the_read)
     }
     __asm__ __volatile__ ("" ::: "memory");
     _private->seqan_spin_lock = 0;
-    if (atEnd) {
+    if (!atEnd) {
+        if (_num_reads > 0 && the_read.quality.length() != 0) {
+            _have_qualities = true;
+        }
+        if (the_read.sequence.length() == 0) {
+            // We decrement the number of reads we've got, as the last one is
+            // invlaid
+            __sync_fetch_and_sub(&_num_reads, 1);
```

Comment:
Hrmm... I'd rather we did all thread-aware operations inside the spinlocks instead of combining them with atomic ops. The check for seq and qual length should therefore go inside the spinlocks. No worse than what will happen when we switch to Seqan 2.0